components/stateTracker.js

```diff
@@ -49,7 +49,7 @@ StateTracker.prototype.add = function(method, newState, url) {
     var stop = false;
     var event = {
         method: method,
-        state: newState,
+        state: _.extend(newState || {}, {_stateFromHistory: false}),
```

Comment:
They are necessary. If you won't do this, _stateFromHistory will always be `true` after first  popstate history.